### ISSUE TITLE
Sometime getting empty list but in console showing contact length received

### ISSUE THREAD

**Issue by Raj-impetrosys** [s0nerik/fast_contacts#25]
See also: When internet is not available not getting contact list but showing in console

I/flutter ( 1257): null
I/flutter ( 1257): Fetched 1215 contacts in 32 ms
I/flutter ( 1257): Received all contacts in 24 ms
![Screenshot 2024-08-07 at 6 13 25 PM](https://github.com/user-attachments/assets/8f1d2268-767a-4ba7-8c9b-05db51266cc6)

**Flutter doctor log**
[✓] Flutter (Channel stable, 3.19.6-0.0.pre.1, on macOS 14.5 23F79 darwin-arm64, locale
    en-IN)
    • Flutter version 3.19.6-0.0.pre.1 on channel stable at
      /Users/rajkumar/Desktop/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 43a3a333d3 (9 weeks ago), 2024-06-06 19:09:30 +0530
    • Engine revision e76c956498
    • Dart version 3.3.3
    • DevTools version 2.31.1

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/rajkumar/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • Java binary at: /Applications/Android
      Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A507
    • CocoaPods version 1.15.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2023.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)

[✓] VS Code (version 1.92.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.94.0

[✓] Connected device (3 available)
    • sdk gphone64 arm64 (mobile) • emulator-5554 • android-arm64  • Android 13 (API 33)
      (emulator)
    • macOS (desktop)             • macos         • darwin-arm64   • macOS 14.5 23F79
      darwin-arm64
    • Chrome (web)                • chrome        • web-javascript • Google Chrome
      127.0.6533.99

[✓] Network resources
    • All expected network resources are available.

• No issues found!

**Comment by s0nerik** [comment]
This sounds like a concurrency issue, but I can't understand how this could happen since `getAllContacts` is guaranteed to be executed sequentially.

Are you sure you received exactly an empty list as a result and not an error?

**Comment by Rajkumar07793** [comment]
> This sounds like a concurrency issue, but I can't understand how this could happen since `getAllContacts` is guaranteed to be executed sequentially.
> 
> Are you sure you received exactly an empty list as a result and not an error?

Yes I'm getting the [] list sometime but printed Fetched 312 contact in console

Which is you printed in flutter code where you defined method channel

See my console i have printed which is null in console, then Fetched 1215 contacts

**Comment by yanashenyang** [comment]
Is there a solution?

**Comment by yanashenyang** [comment]
I also encountered the same problem in iOS

**Comment by s0nerik** [comment]
@yanashenyang do you happen to call `getAllContacts` from multiple isolates?
